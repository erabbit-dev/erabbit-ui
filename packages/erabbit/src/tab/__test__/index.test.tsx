import { mount } from '@vue/test-utils'
import { later } from '../../test'
import { Tabs, TabsPanel } from '..'
import { ref } from 'vue'
import { expect } from 'vitest'

test('should render default Tabs', async () => {
  const wrapper = mount({
    setup() {
      return () => (
        <Tabs>
          <TabsPanel label="首页" name="first">
            1
          </TabsPanel>
          <TabsPanel label="用户管理" name="second">
            2
          </TabsPanel>
          <TabsPanel label="角色管理" name="third">
            3
          </TabsPanel>
        </Tabs>
      )
    },
  })
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
})
test('should render small size Tabs', async () => {
  const wrapper = mount(
    <Tabs size="small">
      <TabsPanel label="首页" name="first">
        1
      </TabsPanel>
      <TabsPanel label="用户管理" name="second">
        2
      </TabsPanel>
      <TabsPanel label="角色管理" name="third">
        3
      </TabsPanel>
    </Tabs>,
  )
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
})
test('should render  mini size Tabs', async () => {
  const wrapper = mount(
    <Tabs size="mini">
      <TabsPanel label="首页" name="first">
        1
      </TabsPanel>
      <TabsPanel label="用户管理" name="second">
        2
      </TabsPanel>
      <TabsPanel label="角色管理" name="third">
        3
      </TabsPanel>
    </Tabs>,
  )
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
  expect(wrapper.find('.er-tabs').exists('.mini')).toBe(true)
})
test('should render card type Tabs', async () => {
  const wrapper = mount(
    <Tabs type="card">
      <TabsPanel label="首页" name="first">
        1
      </TabsPanel>
      <TabsPanel label="用户管理" name="second">
        2
      </TabsPanel>
      <TabsPanel label="角色管理" name="third">
        3
      </TabsPanel>
    </Tabs>,
  )
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
  expect(wrapper.find('.er-tabs').exists('.card')).toBe(true)
})
test('should render border-card type Tabs', async () => {
  const wrapper = mount(
    <Tabs type="border-card">
      <TabsPanel label="首页" name="first">
        1
      </TabsPanel>
      <TabsPanel label="用户管理" name="second">
        2
      </TabsPanel>
      <TabsPanel label="角色管理" name="third">
        3
      </TabsPanel>
    </Tabs>,
  )
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
  expect(wrapper.find('.er-tabs').exists('.border-card')).toBe(true)
})

test('should render left position Tabs', async () => {
  const wrapper = mount(
    <Tabs tabPosition="left">
      <TabsPanel label="首页" name="first">
        1
      </TabsPanel>
      <TabsPanel label="用户管理" name="second">
        2
      </TabsPanel>
      <TabsPanel label="角色管理" name="third">
        3
      </TabsPanel>
    </Tabs>,
  )
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
  expect(wrapper.find('.er-tabs').exists('.left')).toBe(true)
})
test('should render right position Tabs', async () => {
  const wrapper = mount(
    <Tabs tabPosition="right">
      <TabsPanel label="首页" name="first">
        1
      </TabsPanel>
      <TabsPanel label="用户管理" name="second">
        2
      </TabsPanel>
      <TabsPanel label="角色管理" name="third">
        3
      </TabsPanel>
    </Tabs>,
  )
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
})
test('should render bottom position Tabs', async () => {
  const wrapper = mount(
    <Tabs tabPosition="bottom">
      <TabsPanel label="首页" name="first">
        1
      </TabsPanel>
      <TabsPanel label="用户管理" name="second">
        2
      </TabsPanel>
      <TabsPanel label="角色管理" name="third">
        3
      </TabsPanel>
    </Tabs>,
  )
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
  expect(wrapper.find('.er-tabs').exists('.bottom')).toBe(true)
})
test('should render bottom position and card type Tabs', async () => {
  const wrapper = mount(
    <Tabs tabPosition="bottom" type="card">
      <TabsPanel label="首页" name="first">
        1
      </TabsPanel>
      <TabsPanel label="用户管理" name="second">
        2
      </TabsPanel>
      <TabsPanel label="角色管理" name="third">
        3
      </TabsPanel>
    </Tabs>,
  )
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
  expect(wrapper.find('.er-tabs').exists('.bottom')).toBe(true)
  expect(wrapper.find('.er-tabs').exists('.card')).toBe(true)
})
test('should render bottom position and border-card type and mini size Tabs', async () => {
  const wrapper = mount(
    <Tabs tabPosition="bottom" type="border-card" size="mini">
      <TabsPanel label="首页" name="first">
        1
      </TabsPanel>
      <TabsPanel label="用户管理" name="second">
        2
      </TabsPanel>
      <TabsPanel label="角色管理" name="third">
        3
      </TabsPanel>
    </Tabs>,
  )
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
  expect(wrapper.find('.er-tabs').exists('.bottom')).toBe(true)
  expect(wrapper.find('.er-tabs').exists('.border-card')).toBe(true)
})
test('should render bottom position and small size Tabs', async () => {
  const wrapper = mount(
    <Tabs tabPosition="bottom" size="small">
      <TabsPanel label="首页" name="first">
        1
      </TabsPanel>
      <TabsPanel label="用户管理" name="second">
        2
      </TabsPanel>
      <TabsPanel label="角色管理" name="third">
        3
      </TabsPanel>
    </Tabs>,
  )
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
  expect(wrapper.find('.er-tabs').exists('.bottom')).toBe(true)
  expect(wrapper.find('.er-tabs').exists('.small')).toBe(true)
})
test('should render bottom position and mini size Tabs', async () => {
  const wrapper = mount(
    <Tabs tabPosition="bottom" size="mini">
      <TabsPanel label="首页" name="first">
        1
      </TabsPanel>
      <TabsPanel label="用户管理" name="second">
        2
      </TabsPanel>
      <TabsPanel label="角色管理" name="third">
        3
      </TabsPanel>
    </Tabs>,
  )
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)
  expect(wrapper.find('.er-tabs').exists('.bottom')).toBe(true)
  expect(wrapper.find('.er-tabs').exists('.mini')).toBe(true)
})

test('should render active Tabs', async () => {
  let testValue = ''
  const wrapper = mount({
    setup() {
      const activeName = ref('first')
      return () => (
        <Tabs
          modelValue={activeName.value}
          onUpdate:modelValue={(value) => {
            activeName.value = value
            testValue = value
          }}
        >
          <TabsPanel label="首页" name="first">
            1
          </TabsPanel>
          <TabsPanel label="用户管理" name="second">
            2
          </TabsPanel>
          <TabsPanel label="角色管理" name="third">
            3
          </TabsPanel>
        </Tabs>
      )
    },
  })
  await later(100)
  expect(wrapper.html()).toMatchSnapshot()
  expect(wrapper.findAll('.er-tabs-title a').length).toBe(3)

  wrapper.find('.er-tabs-title a:nth-of-type(2)').trigger('click')
  await later()
  expect(testValue).toBe('second')
  wrapper.find('.er-tabs-title a:nth-of-type(3)').trigger('click')
  await later()
  expect(testValue).toBe('third')
  wrapper.find('.er-tabs-title a:nth-of-type(2)').trigger('click')
  await later()
  expect(testValue).toBe('second')
  wrapper.find('.er-tabs-title a:nth-of-type(1)').trigger('click')
  await later()
  expect(testValue).toBe('first')
})
