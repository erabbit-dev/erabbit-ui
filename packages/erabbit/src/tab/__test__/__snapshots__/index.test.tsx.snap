// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render active Tab 1`] = `
"<div class=\\"er-tab er-tab--default er-tab--top\\">
  <div class=\\"er-tab__header\\">
    <div class=\\"er-tab__wrapper\\">
      <div class=\\"er-tab__line\\" style=\\"width: -40px; transform: translateX(20px); transition-duration: 0.3s;\\"></div><a class=\\"er-tab__item is-active\\" href=\\"javascript:;\\">首页</a><a class=\\"er-tab__item\\" href=\\"javascript:;\\">用户管理</a><a class=\\"er-tab__item\\" href=\\"javascript:;\\">角色管理</a>
    </div>
  </div>
  <div class=\\"er-tab__content\\">
    <div class=\\"er-tab-panel is-active\\">1</div>
    <div class=\\"er-tab-panel\\">2</div>
    <div class=\\"er-tab-panel\\">3</div>
  </div>
</div>"
`;

exports[`should render border-card type Tab 1`] = `
"<div class=\\"er-tab er-tab--default er-tab--top er-tab--border-card\\">
  <div class=\\"er-tab__header\\">
    <div class=\\"er-tab__wrapper\\">
      <div class=\\"er-tab__line\\"></div><a class=\\"er-tab__item is-active\\" href=\\"javascript:;\\">首页</a><a class=\\"er-tab__item\\" href=\\"javascript:;\\">用户管理</a><a class=\\"er-tab__item\\" href=\\"javascript:;\\">角色管理</a>
    </div>
  </div>
  <div class=\\"er-tab__content\\">
    <div class=\\"er-tab-panel is-active\\">1</div>
    <div class=\\"er-tab-panel\\">2</div>
    <div class=\\"er-tab-panel\\">3</div>
  </div>
</div>"
`;

exports[`should render bottom position Tab 1`] = `
"<div class=\\"er-tab er-tab--default er-tab--bottom\\">
  <div class=\\"er-tab__header\\">
    <div class=\\"er-tab__wrapper\\">
      <div class=\\"er-tab__line\\"></div><a class=\\"er-tab__item is-active\\" href=\\"javascript:;\\">首页</a><a class=\\"er-tab__item\\" href=\\"javascript:;\\">用户管理</a><a class=\\"er-tab__item\\" href=\\"javascript:;\\">角色管理</a>
    </div>
  </div>
  <div class=\\"er-tab__content\\">
    <div class=\\"er-tab-panel is-active\\">1</div>
    <div class=\\"er-tab-panel\\">2</div>
    <div class=\\"er-tab-panel\\">3</div>
  </div>
</div>"
`;

exports[`should render bottom position and border-card type and small size Tab 1`] = `
"<div class=\\"er-tab er-tab--small er-tab--bottom er-tab--border-card\\">
  <div class=\\"er-tab__header\\">
    <div class=\\"er-tab__wrapper\\">
      <div class=\\"er-tab__line\\"></div><a class=\\"er-tab__item is-active\\" href=\\"javascript:;\\">首页</a><a class=\\"er-tab__item\\" href=\\"javascript:;\\">用户管理</a><a class=\\"er-tab__item\\" href=\\"javascript:;\\">角色管理</a>
    </div>
  </div>
  <div class=\\"er-tab__content\\">
    <div class=\\"er-tab-panel is-active\\">1</div>
    <div class=\\"er-tab-panel\\">2</div>
    <div class=\\"er-tab-panel\\">3</div>
  </div>
</div>"
`;

exports[`should render bottom position and card type Tab 1`] = `
"<div class=\\"er-tab er-tab--default er-tab--bottom er-tab--card\\">
  <div class=\\"er-tab__header\\">
    <div class=\\"er-tab__wrapper\\">
      <div class=\\"er-tab__line\\"></div><a class=\\"er-tab__item is-active\\" href=\\"javascript:;\\">首页</a><a class=\\"er-tab__item\\" href=\\"javascript:;\\">用户管理</a><a class=\\"er-tab__item\\" href=\\"javascript:;\\">角色管理</a>
    </div>
  </div>
  <div class=\\"er-tab__content\\">
    <div class=\\"er-tab-panel is-active\\">1</div>
    <div class=\\"er-tab-panel\\">2</div>
    <div class=\\"er-tab-panel\\">3</div>
  </div>
</div>"
`;

exports[`should render card type Tab 1`] = `
"<div class=\\"er-tab er-tab--default er-tab--top er-tab--card\\">
  <div class=\\"er-tab__header\\">
    <div class=\\"er-tab__wrapper\\">
      <div class=\\"er-tab__line\\"></div><a class=\\"er-tab__item is-active\\" href=\\"javascript:;\\">首页</a><a class=\\"er-tab__item\\" href=\\"javascript:;\\">用户管理</a><a class=\\"er-tab__item\\" href=\\"javascript:;\\">角色管理</a>
    </div>
  </div>
  <div class=\\"er-tab__content\\">
    <div class=\\"er-tab-panel is-active\\">1</div>
    <div class=\\"er-tab-panel\\">2</div>
    <div class=\\"er-tab-panel\\">3</div>
  </div>
</div>"
`;

exports[`should render default Tab 1`] = `
"<div class=\\"er-tab er-tab--default er-tab--top\\">
  <div class=\\"er-tab__header\\">
    <div class=\\"er-tab__wrapper\\">
      <div class=\\"er-tab__line\\"></div><a class=\\"er-tab__item is-active\\" href=\\"javascript:;\\">首页</a><a class=\\"er-tab__item\\" href=\\"javascript:;\\">用户管理</a><a class=\\"er-tab__item\\" href=\\"javascript:;\\">角色管理</a>
    </div>
  </div>
  <div class=\\"er-tab__content\\">
    <div class=\\"er-tab-panel is-active\\">1</div>
    <div class=\\"er-tab-panel\\">2</div>
    <div class=\\"er-tab-panel\\">3</div>
  </div>
</div>"
`;

exports[`should render left position Tab 1`] = `
"<div class=\\"er-tab er-tab--default er-tab--left\\">
  <div class=\\"er-tab__header\\">
    <div class=\\"er-tab__wrapper\\">
      <div class=\\"er-tab__line\\"></div><a class=\\"er-tab__item is-active\\" href=\\"javascript:;\\">首页</a><a class=\\"er-tab__item\\" href=\\"javascript:;\\">用户管理</a><a class=\\"er-tab__item\\" href=\\"javascript:;\\">角色管理</a>
    </div>
  </div>
  <div class=\\"er-tab__content\\">
    <div class=\\"er-tab-panel is-active\\">1</div>
    <div class=\\"er-tab-panel\\">2</div>
    <div class=\\"er-tab-panel\\">3</div>
  </div>
</div>"
`;

exports[`should render right position Tab 1`] = `
"<div class=\\"er-tab er-tab--default er-tab--right\\">
  <div class=\\"er-tab__header\\">
    <div class=\\"er-tab__wrapper\\">
      <div class=\\"er-tab__line\\"></div><a class=\\"er-tab__item is-active\\" href=\\"javascript:;\\">首页</a><a class=\\"er-tab__item\\" href=\\"javascript:;\\">用户管理</a><a class=\\"er-tab__item\\" href=\\"javascript:;\\">角色管理</a>
    </div>
  </div>
  <div class=\\"er-tab__content\\">
    <div class=\\"er-tab-panel is-active\\">1</div>
    <div class=\\"er-tab-panel\\">2</div>
    <div class=\\"er-tab-panel\\">3</div>
  </div>
</div>"
`;

exports[`should render small size Tab 1`] = `
"<div class=\\"er-tab er-tab--small er-tab--top\\">
  <div class=\\"er-tab__header\\">
    <div class=\\"er-tab__wrapper\\">
      <div class=\\"er-tab__line\\"></div><a class=\\"er-tab__item is-active\\" href=\\"javascript:;\\">首页</a><a class=\\"er-tab__item\\" href=\\"javascript:;\\">用户管理</a><a class=\\"er-tab__item\\" href=\\"javascript:;\\">角色管理</a>
    </div>
  </div>
  <div class=\\"er-tab__content\\">
    <div class=\\"er-tab-panel is-active\\">1</div>
    <div class=\\"er-tab-panel\\">2</div>
    <div class=\\"er-tab-panel\\">3</div>
  </div>
</div>"
`;

exports[`should render small size Tab 2`] = `
"<div class=\\"er-tab er-tab--small er-tab--top\\">
  <div class=\\"er-tab__header\\">
    <div class=\\"er-tab__wrapper\\">
      <div class=\\"er-tab__line\\"></div><a class=\\"er-tab__item is-active\\" href=\\"javascript:;\\">首页</a><a class=\\"er-tab__item\\" href=\\"javascript:;\\">用户管理</a><a class=\\"er-tab__item\\" href=\\"javascript:;\\">角色管理</a>
    </div>
  </div>
  <div class=\\"er-tab__content\\">
    <div class=\\"er-tab-panel is-active\\">1</div>
    <div class=\\"er-tab-panel\\">2</div>
    <div class=\\"er-tab-panel\\">3</div>
  </div>
</div>"
`;
